ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -125,4 +166,116 @@ mod tests {
         };
         check(&nd, &mut rng, 1_u8);
     }
+
+    /// Tests for Double Geometric
+    ///
+    #[test]
+    fn test_generate_geometric_sample_dist() {
+        let mut rng = rand::thread_rng();
+        let p = 0.5; // success probability
+        let mut histogram = HashMap::new();
+        let num_samples = 100_000;
+        for _ in 0..num_samples {
+            let sample = generate_geometric(p, &mut rng);
+            *histogram.entry(sample).or_insert(0) += 1;
+        }
+        for x in 0..100 {
+            let observed_probability = histogram
+                .get(&x)
+                .map_or(0.0, |count| f64::from(*count) / f64::from(num_samples));
+            let expected_probability = (1.0 - p).powf(x as f64) * p;
+            // println!("x = {}, Observed Probability = {}, Expected Probability = {}", x, observed_probability, expected_probability);
```

Comment:
you should probably uncomment those, by default cargo test suppresses stdout if test passes. So you won't see noise in your terminal